lib/js/components/Buttons/IconButton/IconButton.vue

```diff
@@ -49,42 +56,120 @@
 
-@mixin iconButtonColor($color, $hover-color: '') {
-	@if $hover-color == '' {
-		//$hover-color: mix($color, $color-firefly-black, (1 - $button-hover-alpha) * 100%);
+@mixin setIconButtonAdditions($ripple: null, $border: null, $icon: null) {
+	@if $ripple != null {
+		.ripple {
+			background-color: $ripple !important;
+		}
 	}
 
-	color: $color;
+	@if $ripple == null {
+		.ripple {
+			display: none;
+		}
+	}
 
-	&:hover,
-	&:active {
-		color: $hover-color;
+	@if $border != null {
+		border: 1px solid $border;
+	}
+
+	@if $icon != null {
+		.a-iconButton {
+			&__icon,
+			&__loadingIcon {
+				color: $icon;
+			}
+		}
 	}
 }
 
 .a-iconButton {
 	$self: &;
 
-	@each $color-name, $color-map in $theme-calculated-colors {
+	@each $color-name, $color-map in $icon-button-colors {
 		&.-color-#{$color-name} {
-			@include iconButtonColor(map-get($color-map, 'color'), map-get($color-map, 'hovered'));
-		}
-	}
+			@include setIconButtonAdditions(
+				map-get($color-map, 'filled', 'ripple'),
+				null,
+				map-get($color-map, 'filled', 'icon')
+			);
 
-	@each $color-name, $color in $regular-colors {
-		&.-color-#{$color-name} {
-			@include iconButtonColor($color);
+			#{$self}__label {
+				color: map-get($color-map, 'outlined', 'color');
+			}
+
+			&:hover,
+			&.-hovered {
+				#{$self}__button {
+					background-color: map-get($color-map, 'filled', 'background-hovered');
+
+					&.-outlined {
+						background-color: map-get($color-map, 'outlined', 'background-hovered');
+						color: map-get($color-map, 'inverted', 'icon');
+					}
+				}
+				#{$self}__label {
+					color: map-get($color-map, 'filled', 'color-hovered');
+				}
+			}
+
+			&:focus,
+			&.-focused {
+				#{$self}__button {
+					background-color: map-get($color-map, 'filled', 'background-focused');
+
+					&.-outlined {
+						background-color: map-get($color-map, 'outlined', 'background-focused');
+						color: map-get($color-map, 'inverted', 'icon');
+					}
+				}
+				#{$self}__label {
+					color: map-get($color-map, 'filled', 'color-focused');
+				}
+			}
+
+			&.-disabled {
+				#{$self}__button {
+					background-color: map-get($color-map, 'filled', 'background-disabled');
+
+					&.-outlined {
+						background-color: map-get($color-map, 'outlined', 'background');
+						border-color: map-get($color-map, 'outlined', 'disabled', 'border');
+
+						.a-iconButton__icon {
+							color: map-get($color-map, 'outlined', 'disabled', 'icon');
+						}
+					}
+				}
+
+				#{$self}__label {
+					color: map-get($color-map, 'outlined', 'disabled', 'color');
+				}
+			}
+
+			.-outlined {
+				@include setIconButtonAdditions(
+					map-get($color-map, 'outlined', 'ripple'),
+					map-get($color-map, 'outlined', 'border'),
+					map-get($color-map, 'outlined', 'icon')
+				);
+			}
 		}
 	}
 
 	align-items: center;
-	color: $color-primary;
+	color: $color-primary-text;
 	cursor: pointer;
 	display: inline-flex;
 	transition: color ease-in-out $default-transition-time;
```

Comment:
Something seems to be overriding this transition time. It feels much longer on master.